website/src/components/CustomCarousel.tsx

```diff
@@ -1,143 +1,103 @@
-        }
-
-        this.showCurrentValue(this.state.items[currentItemId])
-    }
+interface CarouselItem {
+    id: number
+    backgroundClass: string
+    buttonLabel: string
+    headerClass: string
+    text: string | ReactNode
+    itemClass: string
+}
 
-    public render(): JSX.Element | null {
-        return (
-            <div className={this.getBoxClassNames()}>
-                <div className="carousel-nav col-lg-3 col-md-2 ml-lg-7 ml-md-5">
-                    <ArrowUpIcon
-                        className="ml-lg-6 mb-4"
-                        onClick={() => {
-                            this.decrementList()
-                        }}
-                    />
-                    <ul className="ml-lg-3">
-                        {this.props.items.map(item => (
-                            <li
-                                className={item.headerClass}
-                                key={item.id}
-                                onClick={() => {
-                                    this.showCurrentValue(item)
-                                }}
-                            >
-                                {item.buttonLabel}
-                            </li>
-                        ))}
-                    </ul>
-                    <ArrowDownIcon
-                        className="ml-lg-6 mt-4"
-                        onClick={() => {
-                            this.incrementList()
-                        }}
-                    />
-                </div>
-                <div className="col-lg-6 col-md-8 col-sm-12 mt-lg-5 ml-md-6">
-                    {this.props.items.map(item => (
-                        <div key={item.id} className={item.itemClass}>
-                            <h1 className="display-2 mb-lg-4">{item.buttonLabel}</h1>
-                            <p>{item.text}</p>
-                        </div>
+const CustomCarousel: FunctionComponent<CarouselProps> = props => {
+    const carouselLeftPanelStyles = 'carousel-nav col-md-2'
+    const carouselRightPanelStyles = 'col-lg-6 col-md-8 col-sm-12 mt-lg-5 ml-md-6'
+    const carouselMainStyles = 'custom-carousel d-flex flex-wrap'
+    const { items, autoAdvance, backgroundClass } = props
+    const carouselHook = useCarousel(items, autoAdvance ?? false)
+    const carouselItems = carouselHook.carouselItems.items as CarouselItem[]
+
+    return (
+        <div className={classNames(carouselMainStyles, backgroundClass ?? '')}>
+            <div
+                className={
+                    carouselHook.autoAdvance
+                        ? classNames(carouselLeftPanelStyles, 'm-0 col-lg-5')
+                        : classNames(carouselLeftPanelStyles, 'ml-lg-7 col-lg-4 ml-md-5')
+                }
+            >
+                <ArrowUpIcon
+                    className="ml-lg-6 mb-4"
+                    onClick={() => carouselHook.moveCarousel('decrement')}
+                    color={carouselHook.isAdvancing ? '#D0D0D0' : '#000'}
```

Comment:
We are going to update the toggle here to be on opacity, so the "light" arrow shows up on carousels that have gradient backgrounds (like on `/cloud-beta`). I will update this.

Reply:
Done!